lessons/ro/chapter_8.yaml

```
@@ -0,0 +1,357 @@
+  content_markdown: >
+    Pe lângă abilitatea de a crea referințe la tipuri de date existente
+    folosind operatorul `&`, limbajul Rust ne oferă posibilitatea
+    
+    de a crea structuri *asemănătoare cu referințele*  numite **pointeri inteligenți**. 
+
+
+    Ne putem gândi la referințe de nivel înalt ca la un tip ce ne oferă
+    acces la alt tip.  Pointerii inteligenți se comportă diferit față de 
+
+    referințele normale, operând după o logică internă, 
+    scrisă de programator. Tu — programatorul —
+
+    reprezinți partea *inteligentă*.
+
+
+    De obicei pointerii inteligenți implementează trait-urile `Deref`, `DerefMut`, și `Drop` 
+    pentru a specifica logica a ce ar trebui să se întâmple când
+
+    structura este dereferențiată folosind operatorii `*` și `.`.
+- title: Cod nesigur inteligent
+  code: >-
+    https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&code=fn%20main()%20%7B%0A%20%20%20%20let%20a%3A%20%5Bu8%3B%204%5D%20%3D%20%5B86%2C%2014%2C%2073%2C%2064%5D%3B%0A%20%20%20%20%2F%2F%20this%20is%20a%20raw%20pointer.%20Getting%20the%20memory%20address%0A%20%20%20%20%2F%2F%20of%20something%20as%20a%20number%20is%20totally%20safe%0A%20%20%20%20let%20pointer_a%20%3D%20%26a%20as%20*const%20u8%20as%20usize%3B%0A%20%20%20%20println!(%22Data%20memory%20location%3A%20%7B%7D%22%2C%20pointer_a)%3B%0A%20%20%20%20%2F%2F%20Turning%20our%20number%20into%20a%20raw%20pointer%20to%20a%20f32%20is%0A%20%20%20%20%2F%2F%20also%20safe%20to%20do.%0A%20%20%20%20let%20pointer_b%20%3D%20pointer_a%20as%20*const%20f32%3B%0A%20%20%20%20let%20b%20%3D%20unsafe%20%7B%0A%20%20%20%20%20%20%20%20%2F%2F%20This%20is%20unsafe%20because%20we%20are%20telling%20the%20compiler%0A%20%20%20%20%20%20%20%20%2F%2F%20to%20assume%20our%20pointer%20is%20a%20valid%20f32%20and%0A%20%20%20%20%20%20%20%20%2F%2F%20dereference%20it's%20value%20into%20the%20variable%20b.%0A%20%20%20%20%20%20%20%20%2F%2F%20Rust%20has%20no%20way%20to%20verify%20this%20assumption%20is%20true.%0A%20%20%20%20%20%20%20%20*pointer_b%0A%20%20%20%20%7D%3B%0A%20%20%20%20println!(%22I%20swear%20this%20is%20a%20pie!%20%7B%7D%22%2C%20b)%3B%0A%7D%0A
+  content_markdown: >
+    Pointerii inteligenți folosesc adesea cod nesigur . După cum am menționat
+    și mai devreme, 
+
+    aceștia sunt des folosiți în Rust pentru interacțiunea cu memoria la un 
+    nivel de bază.
```

The expression "nivel de bază" could be interpreted as "basic" instead of "close to hardware".
```suggestion
    nivel apropiat de hardware.
```